ssl/internal.h

```diff
@@ -3878,6 +3882,10 @@ struct ssl_ctx_st {
   // of support for AES hardware. The value is only considered if
   // |aes_hw_override| is true.
   bool aes_hw_override_value : 1;
+
+  // State of post handshake authentication. Can be any state from SSL_PHA_STATE
+  uint8_t pha_ext = 0;
```

Comment:
`SSL_CTX` should only be for global configuration settings of the connection. In other words, `SSL_CTX` only controls whether pha is enabled. The actual processing state should be up to `SSL3_STATE` to maintain.